pkg/build/controller/strategy/custom.go

```diff
@@ -104,13 +107,33 @@ func (bs *CustomBuildStrategy) CreateBuildPod(build *buildv1.Build, additionalCA
 					Env:                      containerEnv,
 					SecurityContext:          securityContext,
 					TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts: []corev1.VolumeMount{{
+						Name:      BuildWorkDirVolume,
+						MountPath: buildutil.BuildWorkDirMount,
+					}},
+					// setting the container to run directly in the location the repository is cloned
+					// using the openshift helper
+					WorkingDir: workingDir,
 				},
 			},
+			Volumes: []corev1.Volume{{
+				Name: BuildWorkDirVolume,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			}},
 			RestartPolicy: corev1.RestartPolicyNever,
 			NodeSelector:  build.Spec.NodeSelector,
 		},
 	}
 
+	// when source repository is declared, executing the clone as usual
+	if build.Spec.Source.Git != nil || build.Spec.Source.Binary != nil {
+		setupGitCloneInitContainer(pod, build, bs.Image, containerEnv, securityContext)
+	}
+	// making sure the build working directory is writtable during the process
+	setupChmodInitContainer(pod, build, bs.Image, containerEnv, securityContext, 777, workingDir)
```

Comment:
```suggestion
	setupChmodInitContainer(pod, build, bs.Image, containerEnv, securityContext, 0o777, workingDir)
```
to go with the recent change at util.go:790.